Source/WTF/wtf/HashFunctions.h

```diff
@@ -259,6 +259,17 @@ namespace WTF {
     template<typename T, typename U> struct DefaultHash<std::pair<T, U>> : PairHash<T, U> { };
     template<typename... Types> struct DefaultHash<std::tuple<Types...>> : TupleHash<Types...> { };
 
+    // Default hash for any type with a hash() member function and an equality operator.
+    template<typename T> concept HashableType = requires(const T& t) {
+        { t.hash() } -> std::same_as<unsigned>;
+        { t == t } -> std::same_as<bool>;
+    };
+    template<typename T> requires HashableType<T> struct DefaultHash<T>  {
```

Comment:
Is this another way to write the same thing?
```suggestion
    template<HashableType T> struct DefaultHash<T> [
```